tests/test_memory_core/test_pond.py

```diff
@@ -311,3 +314,98 @@ def test_pond_pe(verilator=True):
         tester.compile_and_run(target=target,
                                tmp_dir=False,
                                **runtime_kwargs)
+
+
+def test_pond_pe_acc(verilator=True):
+
+    chip_size = 2
+    interconnect = create_cgra(chip_size, chip_size, io_sides(),
+                               num_tracks=3,
+                               add_pd=True,
+                               add_pond=True,
+                               mem_ratio=(1, 2))
+
+    netlist = {
+        "e0": [("I0", "io2f_16"), ("p0", "data0")],
+        "e1": [("p0", "data_out_pond"), ("p0", "data1")],
+        "e2": [("p0", "alu_res"), ("p0", "data_in_pond")],
+        "e3": [("p0", "data_out_pond"), ("I1", "f2io_16")]
+    }
+    bus = {"e0": 16, "e1": 16, "e2": 16, "e3": 16}
+
+    placement, routing = pnr(interconnect, (netlist, bus))
+    config_data = interconnect.get_route_bitstream(routing)
+
+    pe_x, pe_y = placement["p0"]
+
+    petile = interconnect.tile_circuits[(pe_x, pe_y)]
+
+    pondcore = petile.additional_cores[0]
+
+    add_bs = petile.core.get_config_bitstream(asm.add())
+    for addr, data in add_bs:
+        config_data.append((interconnect.get_config_addr(addr, 0, pe_x, pe_y), data))
+
+    # Ranges, Strides, Dimensionality, Starting Addr, Starting Addr - Schedule
+    ctrl_rd = [[16, 1], [0, 0], 2, 8, 0, [1, 0]]
+    ctrl_wr = [[16, 1], [0, 0], 2, 8, 0, [1, 0]]
+
+    generate_pond_api(interconnect, pondcore, ctrl_rd, ctrl_wr, pe_x, pe_y, config_data)
+
+    config_data = compress_config_data(config_data)
+
+    circuit = interconnect.circuit()
+
+    tester = BasicTester(circuit, circuit.clk, circuit.reset)
+
+    tester.poke(circuit.interface["stall"], 1)
+
+    for addr, index in config_data:
+        tester.configure(addr, index)
+        tester.config_read(addr)
+        tester.eval()
+        tester.expect(circuit.read_config_data, index)
+
+    tester.done_config()
+    tester.poke(circuit.interface["stall"], 0)
+    tester.eval()
+
+    src_x0, src_y0 = placement["I0"]
+    src_name0 = f"glb2io_16_X{src_x0:02X}_Y{src_y0:02X}"
+    dst_x, dst_y = placement["I1"]
+    dst_name = f"io2glb_16_X{dst_x:02X}_Y{dst_y:02X}"
+    random.seed(0)
+
+    total = 0
+    for i in range(16):
+        tester.poke(circuit.interface[src_name0], i + 1)
+        total = total + i
+        tester.eval()
+        tester.expect(circuit.interface[dst_name], total)
+        tester.step(2)
+        tester.eval()
+
+    with tempfile.TemporaryDirectory() as tempdir:
```

Comment:
Not related to this PR but in the future these code should be refactored and reused.